app/models/permit_application.rb

```diff
@@ -217,6 +207,14 @@ def missing_pdfs
     missing_pdfs
   end
 
+  def energy_step_code_required?
+    jurisdiction
+      .permit_type_required_steps
+      .where(permit_type_id:)
+      .customizations
+      .all? { |rs| !rs.energy_step_required && !rs.zero_carbon_step_required }
```

Comment:
so all of energy_step_required and zero_carbon_step_required have to be false for energy step_code to be required?

Reply:
The real world expectation is that if it's not required, there will only be one customization and both values will be nil. It wouldn't make sense if there are multiple required steps where one of them has both set to not required but if that is the case, we would treat it as required as there is a set of requirements.

Reply:
OK actually yes this logic is flawed, will update

Reply:
@taw27 OK fixed, should be easier to make sense of now and works correctly

Reply:
@LaurelOlson what happens if only one  of `energy_step_required ` or `zero_carbon_step_required ` is nil? This  would be energy_step_code_required? is false?

Reply:
@LaurelOlson approved after latest change